homeassistant/components/netatmo/data_handler.py

```diff
@@ -143,42 +133,47 @@ async def async_fetch_data(self, data_class, data_class_entry, **kwargs):
         except pyatmo.ApiError as err:
             _LOGGER.debug(err)
 
+        except asyncio.TimeoutError as err:
+            _LOGGER.debug(err)
+            return
+
+        for update_callback in self.data_classes[data_class_entry]["subscriptions"]:
+            if update_callback:
+                update_callback()
+
     async def register_data_class(
         self, data_class_name, data_class_entry, update_callback, **kwargs
     ):
         """Register data class."""
-        if data_class_entry in self._data_classes:
-            self._data_classes[data_class_entry]["subscriptions"].append(
-                update_callback
-            )
+        if data_class_entry in self.data_classes:
+            self.data_classes[data_class_entry]["subscriptions"].append(update_callback)
             return
 
-        self._data_classes[data_class_entry] = {
-            "class": DATA_CLASSES[data_class_name],
+        self.data_classes[data_class_entry] = {
             "name": data_class_entry,
             "interval": DEFAULT_INTERVALS[data_class_name],
             NEXT_SCAN: time() + DEFAULT_INTERVALS[data_class_name],
-            "kwargs": kwargs,
             "subscriptions": [update_callback],
         }
 
-        await self.async_fetch_data(
-            DATA_CLASSES[data_class_name], data_class_entry, **kwargs
+        self.data[data_class_entry] = await self.hass.async_add_executor_job(
```

Comment:
Does this need to run in the executor thread pool? Doesn't it just instantiate a class? Are there side effects there?

Reply:
Good point, it doesn't have side effects any more. That is a thing of the past. I'll fix that. Thanks.